includes/plugins/woocommerce/my-account/class-my-account-ui-v1.php

```diff
@@ -24,6 +33,16 @@ public static function init() {
 		\add_action( 'wp_enqueue_scripts', [ __CLASS__, 'enqueue_assets' ], 11 );
 		\add_filter( 'wc_get_template', [ __CLASS__, 'wc_get_template' ], 10, 5 );
 		\add_filter( 'woocommerce_account_menu_items', [ __CLASS__, 'my_account_menu_items' ], 1001 );
+		\add_filter( 'newspack_myaccount_required_fields', [ __CLASS__, 'account_settings_required_fields' ] );
+		\add_action( 'wp_loaded', [ __CLASS__, 'maybe_generate_password_reset_key' ] );
+		\add_action( 'template_redirect', [ __CLASS__, 'redirect_reset_password_link' ], 11 );
+		\add_filter( 'validate_password_reset', [ __CLASS__, 'validate_password_reset' ], 10, 2 );
+		\add_action( 'wp_loaded', [ __CLASS__, 'maybe_password_reset_success' ] );
+		\add_action( 'newspack_woocommerce_after_edit_account_form', [ __CLASS__, 'add_reset_password_modal' ] );
```

Comment:
We have a lot of code just for password reset management. WDYT we put this in a separate class? This is going to grow a lot more as the project progresses.